apis/kueue/webhooks/workload_webhook.go

```diff
@@ -130,6 +132,28 @@ func validatePodSet(ps *kueue.PodSet, path *field.Path) field.ErrorList {
 	for _, msg := range validation.IsDNS1123Label(ps.Name) {
 		allErrs = append(allErrs, field.Invalid(path.Child("name"), ps.Name, msg))
 	}
+
+	// validate initContainers
+	icPath := path.Child("template", "spec", "initContainers")
+	for ci := range ps.Template.Spec.InitContainers {
+		allErrs = append(allErrs, validateContainer(&ps.Template.Spec.InitContainers[ci], icPath.Index(ci))...)
+	}
+	// validate containers
+	cPath := path.Child("template", "spec", "containers")
+	for ci := range ps.Template.Spec.Containers {
+		allErrs = append(allErrs, validateContainer(&ps.Template.Spec.Containers[ci], cPath.Index(ci))...)
+	}
+	return allErrs
+}
+
+func validateContainer(c *corev1.Container, path *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	rPath := path.Child("resources", "requests")
+	for name := range c.Resources.Requests {
+		if name == constants.NumPodsResourceName {
+			allErrs = append(allErrs, field.Invalid(rPath.Key(string(name)), constants.NumPodsResourceName, "the key is reserved for internal kueue use"))
```

Comment:
It doesn't look like kubernetes validates this, but I guess it doesn't hurt to validate https://github.com/kubernetes/kubernetes/blob/51edcd963ec15c7b070c3b504a21fbfe9e978032/pkg/apis/core/validation/validation.go#L5696